### PR TITLE
File/type/conversion/v5

### PR DESCRIPTION
**Pull request by dilanbopanna**

### CODE REVIEW

File: utilities/src/Types/Eventprocessor.res [hunk 1]
```diff
@@ -0,0 +1,4 @@
+type eventProcessor<'a, 'b, 'c> = {
+  processor: (~event: Event.event<'b>, ~hint: Event.eventHint<'c>) => 'a,
+  id?: string, // This field can't be named "name" because functions already have this field natively
```
Comment (prafulkoppalkar) [inline utilities/src/Types/Eventprocessor.res:3]: if sentry has key as name then we cant change it

File: utilities/src/Types/Hub.res [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  bindClient: client => unit, //TODO Client.client
+  @deprecated("Use `withScope` instead.")
+  pushScope: unit => scope, //TODO  Scope.scope
+  @deprecated("Use `withScope` instead.")
+  popScope: unit => bool,
+  withScope: (scope => 't) => 't,
+  getClient: unit => Js.Nullable.t<client>,
+  getScope: unit => scope,
+  getIsolationScope: unit => scope,
+  captureException: (~exception_: 'b, ~hint: Event.eventHint<'a>=?, unit) => string,
```
Comment (prafulkoppalkar) [inline utilities/src/Types/Hub.res:15]: exception_ can this cause arg mismatch ?

**Review by prafulkoppalkar — APPROVED** [review]

File: utilities/src/Types/Eventprocessor.res (hunk 1 above)
Comment (dilanbopanna) [reply utilities/src/Types/Eventprocessor.res:3]: Yeah we can't

File: utilities/src/Types/Hub.res (hunk 2 above)
Comment (dilanbopanna) [reply utilities/src/Types/Hub.res:15]: It wont bcuz in JS order of arguments matters, so the first argument is always exception and second is hint, but if the function captureException is defined with named param
like `({ param2: value2, param1: value1 })`
 then `exception` will go as `undefined` or `null` I guess, So im making this change for safer side

**Pull request merged**